server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -241,7 +271,7 @@ public ClusterTasksResult<Task> execute(ClusterState currentState, List<Task> jo
             }
             results.success(joinTask);
         }
-
+        RepositoriesMetadata repositoriesMetadata = new RepositoriesMetadata(new ArrayList<>(repositories.values()));
```

Comment:
If `repositoriesMetadata` is already being computed above why is it needed to compute again here using repositories map ?

Reply:
The metdata computed above is only for the individual ones (RemoteDN / RemotePublicationDN). Here we are combining the repositories from both the code paths and constructing a new metadata.

Reply:
There is a bit of duplication currently. Can we change to below and then use the repositoriesMetadata instead of created new `repositoriesMetadata`.
```
if (remoteDN.isEmpty() && node.isRemoteStoreNode() || remotePublicationDN.isEmpty() && node.isRemoteStatePublicationEnabled()) {
    repositoriesMetadata = remoteStoreNodeService.updateRepositoriesMetadata(
        node,
        existingrepositoriesMetadata
    );
}   
```
Other is a node is both a remote store and remote publication node, then the logic will be executed twice.

Reply:
Sure makes sense. updated the PR.